src/modules/connection.js

```diff
@@ -71,14 +71,29 @@ class Connection {
 
         // on error we remove the channel so the next call will recreate it (auto-reconnect are handled by connection users)
         channel.on('close', () => { delete connection.chann; });
-        channel.on('error', this._config.transport.error);
+        channel.on('error', this._onError.bind(this));
 
         connection.chann = channel;
         return channel;
       });
     return connection.chann;
   }
 
+  /**
+   * Log errors from connection/channel error events.
+   * @param {Error} error
+   */
+  _onError(error) {
+    this._config.transport.error(error);
+    if (this._config.logger) {
```

Comment:
maybe worth having a `noop` logger by default so you won't need to check for logger defined each time

Reply:
I thouht I could spare the allocation of the log event object this way when the logger is not set. But maybe I overthought it, so should I use a noop function by default?

Reply:
agree with @shamil the overhead is not really matter, it's error logging and not debug logging which can be verbose.